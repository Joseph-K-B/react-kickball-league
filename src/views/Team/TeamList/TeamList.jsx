import {useEffect, useState} from 'react'
import { Link } from 'react-router-dom';
import { deleteTeamById, getTeams } from '../../../services/teams';

function TeamList() {
    const [loading, setLoading] = useState(true);
    const [teams, setTeams] = useState([]);

    const reloadTeams = async () => {
        setLoading(true);
        const res = await getTeams();
        setTeams(res);
        setLoading(false);
    }

    useEffect(() => {
        getTeams()
        .then((res) => setTeams(res))
        .finally(() => setLoading(false))
    }, [])

    const handleDelete = async ({ id, name }) => {
        // eslint-disable-next-line no-restricted-globals
        const confirmDelete = confirm(`Would you like to remove ${name} from this seasons roster?`);

        if(confirmDelete) {
            await deleteTeamById(id);
            await reloadTeams();
        }
    };

    if(loading) return <h1>Loading teams...</h1>
    
    return (
        <>
        <h1>Team List</h1>
        <ul>
            {teams.map((team) => {
                return (
                    <li key={team.id}>
                        <Link className='link' to={`/teams/${team.id}`}>
                        <p>{team.name}</p>
                        </Link>
                        <button
                            type="button"
                            aria-label={`Delete ${team.name}`}
                            onClick={() => handleDelete({ id: team.id, name: team.name})}
                        >
                            Delete
                        </button>
                        <Link to={`/teams/${team.id}/update`}>
                        <button
                            type="button"
                            aria-label={`Update ${team.name}`}
                        >
                            Update
                        </button>
                        </Link>
                    </li>
                    )
                })}
            </ul>            
            <Link className='link' to='/teams/new'><p>Add Team</p></Link>
            </>
    )
}

export default TeamList