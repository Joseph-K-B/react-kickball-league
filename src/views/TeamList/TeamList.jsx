import {useEffect, useState} from 'react'
import { Link } from 'react-router-dom';
import { getTeams } from '../../services/teams';

function TeamList() {
    const [loading, setLoading] = useState(true);
    const [teams, setTeams] = useState([]);

    useEffect(() => {
        getTeams()
        .then((res) => setTeams(res))
        .finally(() => setLoading(false))
    }, [])

    if(loading) return <h1>Loading teams...</h1>
    
    return (
        <>
        <h1>Team List</h1>
        <ul>
            {teams.map((team) => (
                <li key={team.id}>
                    <Link className='link' to={`/teams/${team.id}`}>
                    <p>{team.name}</p>
                    </Link>
                </li>
            ))}
        </ul>
        <Link className='link' to='/teams/new'><p>Add Team</p></Link>
        </>
    )
}

export default TeamList