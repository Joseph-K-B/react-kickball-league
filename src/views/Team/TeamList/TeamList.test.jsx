import { render, screen } from '@testing-library/react'
import { setupServer } from 'msw/node';
import { rest } from 'msw';
import { MemoryRouter, Route } from 'react-router-dom'
import TeamList from './TeamList';

const mockTeam1 = {
    id: 1,
    created_at: '',
    name: 'redirect',
    city: 'random',
    state: 'Vatican City',
    players: []
};

const mockTeam2 = {
    id: 2,
    created_at: '',
    name: 'another one',
    city: 'random',
    state: 'curious',
    players: []
};

const mock = [mockTeam1, mockTeam2]

const server = setupServer(
    rest.get('url', (req, res, ctx) => {
        return res(
            ctx.json(mock)
        );
    }),
    rest.delete('url', (req, res, ctx) => {
        return res(ctx.json(mockTeam2));
    })
);

beforeAll(() => {
    server.listen();
});

afterAll(() => {
    server.close();
});

it('renders team details according to id param', async () => {
    const {container} = render(
        <MemoryRouter initialEntries={['/teams']}>
            <Route path="/teams">
                <TeamList />
            </Route>
        </MemoryRouter>
    );

    screen.getByText('Loading teams...');
    await screen.findByText(/Add Team/);

    const team = await screen.findByText(/Bridge City Sinners/, {exact: false});
    const teamTwo = await screen.findByText(/Seattelites/, {exact: false});
    const teamThree = await screen.findByText(/Mt. hoodRats/, {exact: false});
    expect(team).toBeInTheDocument();
    expect(teamTwo).toBeInTheDocument();
    expect(teamThree).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})