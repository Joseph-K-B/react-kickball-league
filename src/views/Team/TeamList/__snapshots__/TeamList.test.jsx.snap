// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders team details according to id param 1`] = `
<div>
  <h1
    class="listTitle"
  >
    Teams
  </h1>
  <ul
    class="teamList"
  >
    <li>
      <a
        class="link"
        href="/teams/1"
      >
        <p>
          Bridge City Sinners
        </p>
      </a>
      <button
        aria-label="Delete Bridge City Sinners"
        type="button"
      >
        Delete
      </button>
      <a
        href="/teams/1/update"
      >
        <button
          aria-label="Update Bridge City Sinners"
          type="button"
        >
          Update
        </button>
      </a>
    </li>
    <li>
      <a
        class="link"
        href="/teams/3"
      >
        <p>
          Mt. hoodRats
        </p>
      </a>
      <button
        aria-label="Delete Mt. hoodRats"
        type="button"
      >
        Delete
      </button>
      <a
        href="/teams/3/update"
      >
        <button
          aria-label="Update Mt. hoodRats"
          type="button"
        >
          Update
        </button>
      </a>
    </li>
    <li>
      <a
        class="link"
        href="/teams/7"
      >
        <p>
          Neverhalem
        </p>
      </a>
      <button
        aria-label="Delete Neverhalem"
        type="button"
      >
        Delete
      </button>
      <a
        href="/teams/7/update"
      >
        <button
          aria-label="Update Neverhalem"
          type="button"
        >
          Update
        </button>
      </a>
    </li>
    <li>
      <a
        class="link"
        href="/teams/4"
      >
        <p>
          Once More unto the Cannon Breach
        </p>
      </a>
      <button
        aria-label="Delete Once More unto the Cannon Breach"
        type="button"
      >
        Delete
      </button>
      <a
        href="/teams/4/update"
      >
        <button
          aria-label="Update Once More unto the Cannon Breach"
          type="button"
        >
          Update
        </button>
      </a>
    </li>
    <li>
      <a
        class="link"
        href="/teams/2"
      >
        <p>
          PDX-WIK
        </p>
      </a>
      <button
        aria-label="Delete PDX-WIK"
        type="button"
      >
        Delete
      </button>
      <a
        href="/teams/2/update"
      >
        <button
          aria-label="Update PDX-WIK"
          type="button"
        >
          Update
        </button>
      </a>
    </li>
    <li>
      <a
        class="link"
        href="/teams/8"
      >
        <p>
          Salem Pitch Trials
        </p>
      </a>
      <button
        aria-label="Delete Salem Pitch Trials"
        type="button"
      >
        Delete
      </button>
      <a
        href="/teams/8/update"
      >
        <button
          aria-label="Update Salem Pitch Trials"
          type="button"
        >
          Update
        </button>
      </a>
    </li>
    <li>
      <a
        class="link"
        href="/teams/5"
      >
        <p>
          Seattelites
        </p>
      </a>
      <button
        aria-label="Delete Seattelites"
        type="button"
      >
        Delete
      </button>
      <a
        href="/teams/5/update"
      >
        <button
          aria-label="Update Seattelites"
          type="button"
        >
          Update
        </button>
      </a>
    </li>
    <li>
      <a
        class="link"
        href="/teams/6"
      >
        <p>
          Six Quadrants
        </p>
      </a>
      <button
        aria-label="Delete Six Quadrants"
        type="button"
      >
        Delete
      </button>
      <a
        href="/teams/6/update"
      >
        <button
          aria-label="Update Six Quadrants"
          type="button"
        >
          Update
        </button>
      </a>
    </li>
  </ul>
  <a
    class="link"
    href="/teams/new"
  >
    <p>
      Add Team
    </p>
  </a>
</div>
`;
